src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomGeneratorSettings.java

```diff
@@ -301,237 +375,37 @@ public static CustomGeneratorSettings fromVanilla(ChunkGeneratorSettings setting
-                    ));
-                }
-                if (!root.has("cubeAreas")) {
-                    root.add("cubeAreas", new JsonObject());
-                }
-                if (!root.has("replacerConfig")) {
-                    JsonObject replacerConf = new JsonObject();
-                    {
-                        JsonObject defaults = new JsonObject();
-                        {
-                            defaults.add(MODID + ":horizontal_gradient_depth_decrease_weight", new JsonPrimitive(1.0f));
-                            defaults.add(MODID + ":height_offset", new JsonPrimitive(offset));
-                            JsonObject terrainfill = new JsonObject();
-                            {
-                                JsonObject properties = new JsonObject();
-                                properties.add("variant", new JsonPrimitive("stone"));
-                                terrainfill.add("Properties", properties);
-                                terrainfill.add("Name", new JsonPrimitive("minecraft:stone"));
-                            }
-                            JsonObject oceanblock = new JsonObject();
-
-                            {
-                                JsonObject properties = new JsonObject();
-                                properties.add("level", new JsonPrimitive("0"));
-                                oceanblock.add("Properties", properties);
-                                oceanblock.add("Name", new JsonPrimitive("minecraft:water"));
-                            }
-                            defaults.add(MODID + ":biome_fill_depth_offset", new JsonPrimitive(3.0f));
-                            defaults.add(MODID + ":biome_fill_noise_octaves", new JsonPrimitive(4.0f));
-                            defaults.add(MODID + ":height_scale", new JsonPrimitive(factor));
-                            defaults.add(MODID + ":biome_fill_depth_factor", new JsonPrimitive(2.3333333333333335f));
-                            defaults.add(MODID + ":mesa_depth", new JsonPrimitive(16.0f));
-                            defaults.add(MODID + ":water_level", root.get("waterLevel"));
-                            defaults.add(MODID + ":biome_fill_noise_freq", new JsonPrimitive(0.0078125f));
-                        }
-                        replacerConf.add("defaults", defaults);
-                        replacerConf.add("overrides", new JsonObject());
-                    }
-
-                    root.add("replacerConfig", replacerConf);
-                }
-                compound.setString("generatorOptions", gson.toJson(root));
-                return compound;
-            }
-        });
+    public static String fixGeneratorOptions(String generatorOptionsToFix) {
+        generatorOptionsToFix = generatorOptionsToFix.replaceAll("cubicchunks:", MODID + ":");
+        String newGeneratorOptions = new CustomGeneratorSettings().toJson(true);
```

Comment:
That's not how it should be. Fixer behavior should NOT change once the `CustomGeneratorSettings` class is changed.
Notice that I only used it in the last datafixer, and it was done only to make sure the generator settings string is nonempty. This was not documented, but my intention on this was to always move that part to the current last data fixer.